arm11/source/start.s

```diff
@@ -127,4 +142,19 @@ wait_arm9:
 	@ Jump to the kernel!
 	bx lr
 
+smp_boot:
+	@ r0 is the current CPU ID
+	ldr r1, =SYNC_ADDR
+	add r1, r1, r0, lsl #2
+	mov r12, #0
+	str r12, [r1]
```

Comment:
wont core 0 also be trying to write a `0` to `SYNC_ADDR`?  So isn't there a data race here?

Imagine:
1. core 0 is in the `wait_arm9` loop, waiting to read non-zero from `SYNC_ADDR`
2. core 1 is in `smp_boot`.
3. arm9 writes to SYNC_ADDR
4. core 1 writes a `0` to `SYNC_ADDR` in smp_boot then advances to wait_smpboot loop waiting to read non-zero from `SYNC_ADDR`.

So both cores miss the update from the arm9?  Am I misunderstanding? Perhaps a second address is needed for the second core to get notified by the first arm11 (rather than the arm9)?

Reply:
the thing is core1 will write a 0 to (SYNC_ADDR + 1*4), which is it's own "smp boot address" so to speak, the last 16 bytes of AXI WRAM are reserved for us to do that

Reply:
ah, no, the additional cores are using SYNC_ADDR + CPUID << 2, not SYNC_ADDR which cpu 0 is.